File: python/triangle/triangle.py
def is_triangle(sides):
    a,b,c = sides
    return a > 0 and a + b >= c and b + c >= a and a + c >= b

def equilateral(sides):
    if not is_triangle(sides): return False
    a, b, c = sides
    return a == b == c


def isosceles(sides):
    if not is_triangle(sides): return False
    a, b, c = sides
    return a == b or b == c or a == c


def scalene(sides):
    if not is_triangle(sides): return False
    a, b, c = sides
    return a != b != c and a != c

File: ruby/triangle/triangle.rb
class Triangle
  def initialize(sides)
    @sides = sides
  end
  def triangle?
    @sides.each {|s| return false if s == 0 }
    (@sides[0] + @sides[1]) >= @sides[2] &&
    (@sides[0] + @sides[2]) >= @sides[1] &&
    (@sides[1] + @sides[2]) >= @sides[0]
  end

  def equilateral?
    triangle? and (@sides.uniq.size == 1)
  end
  def isosceles?
    triangle? and (@sides.uniq.size < 3)
  end
  def scalene?
    triangle? and (@sides.uniq.size == 3)
  end
end
